### ISSUE TITLE
Question in code

### ISSUE THREAD

**Issue by JiangCe0810** [eriklindernoren/PyTorch-YOLOv3#56]
In the method build_target of /utils/utils.py, why do we set conf_mask is ones rather than zero?
And what is the code 'conf_mask[b, anch_ious > ignore_thres] = 0' meaning?
Thanks

**Comment by khurramHazen** [comment]
Same question.
BDW my guess is author intended to implement "If the bounding box prior is  not  the  best  but  does  overlap  a  ground  truth  object  by more than some threshold we ignore the prediction, following [17]. We use the threshold of 0.5 (YOLOv3 Tech Report)" . If this was his/her intention then it's not correct implementation.

**Comment by khurramHazen** [comment]
@JiangCe0810 If you have reviewed this code can you please comment on following two points:
1. The paper seems to suggest to use MSE loss for box coordinate prediction and BCE for confidence prediction. So there is a bug in models.py line # 162-163. Right? Or am I missing something in paper?
2. There does not seem to be any guarantee that a single ground truth box does not get assigned to two anchors in two different levels of yolo layers. Right?

**Comment by ydixon** [comment]
It seems everyone has their own take on how to implement the loss function. There's few yolov3 implementation on Github, and they are all slightly different. My guess is that they all converge but maybe in different speed.

**Comment by WormCoder** [comment]
@khurramHazen you might confuse the prior box with the predicted box. I think 'anch_ious > ignore_thres' is reasonable.

**Comment by eriklindernoren** [comment]
Hi, let's keep training issues in #2.
